lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2533,7 +2671,7 @@ def main():
         folder=dict(type='str'),
         guest_id=dict(type='str'),
         disk=dict(type='list', default=[]),
-        cdrom=dict(type='dict', default={}),
+        cdrom=dict(type=list_or_dict, default=[]),
```

Comment:
@Akasurde is there an established pattern in Ansible Core to handle this kind of field validation?

Reply:
I saw this method in "vmware_deploy_ovf", so I defined "def list_or_dict(value)" at the top of this module.